index.d.ts

```diff
@@ -0,0 +1,304 @@
+  }
+  interface DeleteOutput<Schema> extends Omit<DeleteItemOutput, "Attributes"> {
+    Attributes?: Schema & SchemaBase;
+  }
+  interface UpdateOutput<Schema> extends Omit<UpdateItemOutput, "Attributes"> {
+    Attributes?: Schema & SchemaBase;
+  }
+
+  export class Entity<
+    Schema extends { [key in keyof Schema]: SchemaType },
+    HiddenKeys extends Omit<Partial<string>, keyof Schema>
+  > {
+    constructor(options: EntityConstructor<Schema, HiddenKeys>);
+
+    // Properties
+    table: Table;
+    readonly DocumentClient: DocumentClient;
+    autoExecute: boolean;
+    autoParse: boolean;
+    readonly partitionKey: string;
+    readonly sortKey: string;
+
+    // Methods
+    attribute(attribute: string): string;
+    parse(input, include?);
+    get(
+      item: Schema,
+      options?: {
+        consistent?: boolean;
+        capacity?: NoneOption | TotalOption | IndexesOption;
+        attributes?: [] | {};
+        autoExecute?: boolean;
+        autoParse?: boolean;
+      },
+      parameters?
+    ): Promise<GetOuput<Schema>>;
+    delete(
+      key: {},
+      options?: {
+        conditions?: [] | {};
+        capacity?: NoneOption | TotalOption | IndexesOption;
+        metrics?: NoneOption | SizeOption;
+        returnValues?: NoneOption | AllOldOption;
+        autoExecute?: boolean;
+        autoParse?: boolean;
+      },
+      parameters?
+    ): Promise<DeleteOutput<Schema>>;
+    put(
+      item: Schema,
+      options?: {
+        conditions?: [] | {};
+        capacity?: NoneOption | TotalOption | IndexesOption; // Cannot use DynamoDB type as it's upper-case
+        metrics?: NoneOption | SizeOption;
+        returnValues?: NoneOption | AllOldOption;
+        autoExecute?: boolean;
+        autoParse?: boolean;
+      },
+      parameters?
+    ): Promise<PutOutput<Schema>>;
+    update(
+      key: Schema & { $remove?: string[] }, // Complex part to type
+      options?: {
+        conditions?: [] | {};
+        capacity?: NoneOption | TotalOption | IndexesOption;
+        metrics?: NoneOption | SizeOption;
+        returnValues?:
+          | NoneOption
+          | AllOldOption
+          | UpdatedOldOption
+          | AllNewOption
+          | UpdatedNewOption;
+        autoExecute?: boolean;
+        autoParse?: boolean;
+      },
+      parameters?
+    ): Promise<UpdateOutput<Schema>>;
+    query(
+      partitionKey: string,
+      options?: QueryInputOptions,
+      parameters?
+    ): Promise<any>;
+    scan(options?, parameters?);
+  }
+
+  // Table
+  interface TableAttributes {
+    [attribute: string]: AnyDynamoDBType;
+  }
+  interface Indexes {
+    [index: string]: { partitionKey?: string; sortKey?: string };
+  }
+  interface TableConstructor {
+    name: string;
+    alias?: string;
+    partitionKey: string;
+    sortKey?: string;
+    entityField?: boolean | string;
+    attributes?: TableAttributes;
+    indexes?: Indexes;
+    autoExecute?: boolean;
+    autoParse?: boolean;
```

Comment:
```suggestion
    autoParse?: boolean;
    removeNullAttributes?: boolean; // Removes null attributes instead of setting them to null (default: true)
```